examples/bert/run_pretraining.py

```diff
@@ -398,16 +421,32 @@ def main(_):
         num_warmup_steps=FLAGS.num_warmup_steps,
         num_train_steps=FLAGS.num_train_steps,
     )
+    optimizer = keras.optimizers.Adam(learning_rate=learning_rate_schedule)
+
+    if FLAGS.skip_restore and len(os.listdir(FLAGS.checkpoint_save_dir)) > 0:
```

Comment:
This hits some weird undefined behavior if `skip_restore=True` and `checkpoint_save_dir=None`. Maybe something like

```
if FLAGS.skip_restore and FLAGS.checkpoint_save_dir is not None:
    if os.path.isdir(FLAGS.checkpoint_save_dir):
         rmdir...
    else:
        raise error should be directory
```
```

Reply:
Yes, we should check if it is a dir before deleting it, but we probably don't want to error out when it's not a directory? If it's not a directory they can still write to the path. My concern is that we are exposing a strange logic - if you want to skip restoring, you have to make sure that `checkpoint_save_directory` either points to nothing or a directory. 

Updated the code with directory check, and we can discuss on if error is needed here.

Reply:
I'm not sure the use case you are trying to protect? There is never a way a user can write to a file.

In the version I suggested, if checkpoint dir is a file, you get an error right away. In the version you pushed, you just get the error after the first epoch from deeper withing tensorflow `tensorflow.python.framework.errors_impl.FailedPreconditionError: /home/matt/bert_test_output/myfile.ckpt is not a directory`.

Thinking about it more, it is probably just be nicer to give that warning regardless.

```
if FLAGS.checkpoint_save_dir is not None and os.path.exists(FLAGS.checkpoint_save_dir):
    if not os.path.isdir(FLAGS.checkpoint_save_dir):
        raise error should be directory
    elif FLAGS.skip_restore:
        rmdir
```